shopelectro/selenium/elements/cart.py

```diff
@@ -22,20 +25,33 @@ def _hover(self):
             (By.CLASS_NAME, 'js-cart-wrapper')
         ))
 
+    @contextmanager
+    def wait_changes(self):
+        def wait_changes(browser):
+            try:
+                return positions_before != self.positions()
+            except TimeoutException:
+                return False
+
+        positions_before = self.positions()
+        yield
+        self.driver.wait.until(wait_changes)
+
     def positions(self) -> [elements.CartPosition]:
-        positions_count = len(self.driver.find_elements(
-            By.CLASS_NAME, 'basket-item'
-        ))
+        try:
+            # use short_wait to avoid long pauses in case of the empty cart
+            positions_count = len(self.driver.short_wait.until(EC.presence_of_all_elements_located(
+                (By.CLASS_NAME, 'basket-item')
+            )))
+        except TimeoutException:
+            return []
```

Comment:
```suggestion
            positions_count = 0
```

it would be more idiomatic, but it's up to you